src/utils/DMRoomMap.ts

```diff
@@ -207,13 +208,22 @@ export default class DMRoomMap {
 
     public getUniqueRoomsWithIndividuals(): { [userId: string]: Room } {
         if (!this.roomToUser) return {}; // No rooms means no map.
-        return Object.keys(this.roomToUser)
-            .map((r) => ({ userId: this.getUserIdForRoomId(r), room: this.matrixClient.getRoom(r) }))
-            .filter((r) => r.userId && r.room?.getInvitedAndJoinedMemberCount() === 2)
-            .reduce((obj, r) => {
-                obj[r.userId] = r.room;
-                return obj;
-            }, {} as Record<string, Room>);
+        return filterBoolean<{ userId: string; room: Room }>(
```

Comment:
The filter + map + reduce could be packed into a single reduce.
It does now iterate three times over all the values.

Reply:
Good point